### PR TITLE
update readme and updt __init__

### PR DESCRIPTION
**Pull request by vchiley**
read through the llmfoundry and scripts/train README's
updated llmfoundry's `__int__.py` to import optim and utils

### CODE REVIEW

File: scripts/train/README.md [hunk 1]
```diff
@@ -117,19 +117,19 @@ composer --world_size 16 --node_rank 1 --master_addr 0.0.0.0 --master_port 7501
...
 # export MASTER_ADDR=0.0.0.0
 # export MASTER_PORT=7501
-composer train.py yamls/mosaic_gpt/125m.yaml
+composer train.py yamls/mpt/125m.yaml
 ```
 
 You should see logs being printed to your terminal like so.
 You can also easily enable other experiment trackers like Weights and Biases or CometML,
-by using [Composer's logging integrations](https://docs.mosaicml.com/en/stable/trainer/logging.html).
+by using [Composer's logging integrations](hhttps://docs.mosaicml.com/projects/composer/en/latest/trainer/logging.html).
```
Comment (dakinggg) [inline scripts/train/README.md:132]: typo

File: scripts/train/README.md [hunk 2]
```diff
@@ -54,11 +54,11 @@ To verify that the dataloader works, run a quick test on your `val_small` split
 # This will construct a `StreamingTextDataset` dataset from your `val` split,
 # pass it into a PyTorch Dataloader, and iterate over it and print samples.
 # Since we only provide a local path, no streaming/copying takes place.
-python ../data_prep/text_data.py --local_path ./my-copy-c4 --split val_small
+python ../../llmfoundry/data/text_data.py --local_path ./my-copy-c4 --split val_small
```
Comment (alextrott16) [inline scripts/train/README.md:57]: Can you double check that the extra `../` is necessary?

File: scripts/train/README.md [hunk 3]
```diff
@@ -346,4 +346,4 @@ In previous blogs ([1](https://www.mosaicml.com/blog/farewell-oom), [2](https://
 we also demonstrated Auto Grad Accum, which takes things a step further by letting Composer determine the `device_train_microbatch_size` on its own.
 This makes our configs not only device-count-agnostic, but hardware-agnostic too!
 You can try out this feature by setting `device_train_microbatch_size: auto`, but bear in mind that FSDP support is still in alpha mode
-and may not always work with Auto Grad Accum (but we are working on it!).
+and may not always work with Auto Grad Accum (we are working on it!).
```
Comment (dakinggg) [inline scripts/train/README.md:349]: auto microbatchin not auto grad accum

**Review by dakinggg — APPROVED** [review]

**Review by alextrott16 — APPROVED** [review]
The selected components mentioned in `llmfoundry/README.md` seem like a bit of an odd collection. I think it's fine for now, but as a future TODO we may want to flesh this out a bit and/or organize it a bit more.

File: scripts/train/README.md (hunk 2 above)
Comment (vchiley) [reply scripts/train/README.md:57]: yeah if you're in `llm-foundry/scripts/train/` and want to access `llm-foundry/llmfoundry/` you need to go up twice.

**Pull request merged**